### PR TITLE
Add Dataset to RIS

### PR DESCRIPTION
**Pull request by adam3smith**
Also -- lint...

### CODE REVIEW

**Comment by adam3smith** [comment]
Question while I'm working on this @dstillman @AbeJellinek  -- There are ~20 `Z.debug` statements in here that seem to slow import down noticeably (going by how quickly the tests in Scaffold run with and without). I don't actually think they help us catch any issues on import, so would it be OK (and, I would argue, desirable, since people use RIS for large import a lot) to comment them out? Or are they only doing anything when debug output is enabled (in which case speed would matter much less).

File: RIS.js [hunk 1]
```diff
@@ -57,83 +81,83 @@ var DEFAULT_EXPORT_TYPE = 'GEN';
...
+	blogPost: "BLOG",
+	book: "BOOK",
+	bookSection: "CHAP",
+	case: "CASE",
+	computerProgram: "COMP",
+	conferencePaper: "CONF",
+	dictionaryEntry: "DICT",
+	encyclopediaArticle: "ENCYC",
+	email: "ICOMM",
+	dataset: "DATA",
```
Comment (adam3smith) [inline RIS.js:96]: Note -- this doesn't check if dataset is available in the importing Zotero version: I didn't know how to do that here in the object and without that it'd be a mess to hack into the translator

Comment (adam3smith) [reply RIS.js:96]: No, that's not a mistake. `ExportTypeMap` holds only bi-directional, unique mappings, and [is merged with](https://github.com/zotero/translators/blob/246071dd68bc3d7bf068cb8afff63e19f227408e/RIS.js#L165) `ImportTypeMap` (see also the comments above each of them).
So this does impact import -- could we do the item type switch all the way at the end in post-processing? That seems cleanest, I think.

Comment (adam3smith) [reply RIS.js:96]: Cool, actually put the code all the way at the top so the item type would be treated correctly during import throughout. Works nicely best I could test.

**Comment by dstillman** [comment]
Slowdown is due to [this line](https://github.com/zotero/zotero/blob/7473515d53489d0acd2216ab91851be45aa80696/chrome/content/scaffold/scaffold.js#L1147).

**Comment by adam3smith** [comment]
@AbeJellinek lmk when/if this is good to merge.

**Pull request merged**